Feat/add graph visualization

Apoorva64 opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

### Description
This PR refactors some parts of the Live trace component to make its inner workings reusable. 
This PR adds the graph visualization

https://github.com/user-attachments/assets/283cb79f-1960-4cad-9faf-4455aac730da



This PR is still in draft as we still need to cleanup some parts of the code
### Related issue(s)
#1700

lbroudoux commented on webapp/src/main/webapp/src/app/components/live-traces/live-traces-graph/live-traces-graph.component.css:
```diff
@@ -0,0 +1,125 @@
+.graph-info {
+  margin-bottom: 12px;
+  padding-bottom: 12px;
+  border-bottom: 2px solid #eee;
```
Could you use CSS color variables from `styles.css` here?

lbroudoux commented on webapp/src/main/webapp/src/app/components/live-traces/live-traces-graph/live-traces-graph.component.css:
```diff
@@ -0,0 +1,125 @@
+  margin: 0 0 8px 0;
+  font-size: 16px;
+  font-weight: 600;
+  color: #333;
```
Could you use CSS color variables from `styles.css` here?

lbroudoux commented on webapp/src/main/webapp/src/app/components/live-traces/live-traces-graph/live-traces-graph.component.html:
```diff
@@ -0,0 +1,69 @@
+            fill="none"
+            [attr.d]="ctx.path()"
+            [attr.stroke-width]="2"
+            [attr.stroke]="'#666'"
```
Could you use CSS color variables from `styles.css` here?

Apoorva64 commented:
@lbroudoux i am not too sure about the colors and the titles and the subtitles. i've fixed some interactions and added the colors for dark mode. The colors for the graph are in a section in style.css

https://github.com/user-attachments/assets/27466391-7ad1-4316-b77b-f13093b8e750

lbroudoux reviewed (approved):
It looks great and all good to me! Thanks a ton @Apoorva64!

lbroudoux commented:
Let merge this one and we'll be ready for `1.13.0` release!!
